test-plan.md

```diff
@@ -0,0 +1,100 @@
+
+Project types to test:
+* Standalone csproj 
+* Directory containing .sln file that references csprojs--projects should be loaded 
+* .NET Core/.NET Standard csproj
+* (Windows) Desktop .NET projects
+* Unity projects
+
+The easist way to verify that a project was successfully loaded is to open a .cs file within it and verify that the references codelens indicator appears.
+
+#### Specific projects to test opening (on all OSes):
+* `dotnet new console` in a directory
+* A more complex dotnet solution, eg. [1] 
+* A console app solution created in Visual Studio
+
+#### Unity
+* Follow the directions at https://code.visualstudio.com/docs/other/unity to configure VS Code with unity
+* Try editing a project like https://github.com/staceyhaffner/SuperSpaceShooter
+
+#### Intellisense 
+* The completion list in a file contains symbols defined in references and in the file
+* If you write a documentation comment on a symbol, the completion list displays it
+
+#### Signature Help 
+* Signature Help shows up in a method call after typing `(`
+* Signature help shows documentation for methods and for parameters
+
+#### Quick Info
+* Hovering over an identifier shows info and documentation
+
+#### Formatting
+* The "Format Document" command works
+* Pressing enter inside a method body automatically indents the new line
+
+#### Go To Definition
+* F12 from callsites to definition
+* Ctrl-Click
+* Can go to metadata for symbols defined in metadata
+
+#### Go To Implementation
+* Ctrl-F12 on virtual member shows implementations
+
+#### Find All References
+* Shift-F12 on a symbol shows a reference list
+
+### Reference highlighting
+* Clicking on a symbol name should highlight other references to it in the same file
+
+#### Colorization
+* Appropriate colorization of keywords, literals, identifiers
+
+#### Error squiggles
+* Introducing an error should show a squiggle in the editor and an error message in the "problems" window
+* Messages in the "problems" window should appear only once per unique error (see https://github.com/OmniSharp/omnisharp-vscode/issues/1830)
+
+#### Quick Fixes
+* Add using should be availble (does not currently suppport adding references)
+* Generate variable/generate method should show up for missing members
+* Remove unncessary usings should show up
+* (this is not an exhaustive list)
+
+#### Refactorings
+* `Use expression body for methods` should be available
+* `Rename file to match type` should correctly rename the file. The renamed file should be open after the refactoring
+* (this is not an exhaustive list)
+
+#### Code Lens - References
+* References codelens appears on symbols and shows correct Find All References results
+
+#### Code Lens - Unit Tests
+* In unit tests projects, the "run test" and "debug test" codelens appears on test methods
+  * Clicking runs or debugs the test and prints results to the console
+  * Breakpoints in a unit test are hit if you "debug test"
+
```

Comment:
@rchande Should we mention about "run all tests" and "debug all tests" codelens that was recently added (https://github.com/OmniSharp/omnisharp-vscode/pull/1961) ?

Reply:
Done